tests/fail-dep/concurrency/libc_pthread_mutex_deadlock.stderr

```diff
@@ -6,6 +6,19 @@ LL |             assert_eq!(libc::pthread_mutex_lock(lock_copy.0.get() as *mut _
    |
    = note: BACKTRACE on thread `unnamed-ID`:
    = note: inside closure at tests/fail-dep/concurrency/libc_pthread_mutex_deadlock.rs:LL:CC
+   = note: thread `unnamed-ID` was spawned by thread `main`
+   = note: inside `std::sys::pal::PLATFORM::thread::Thread::new` at RUSTLIB/std/src/sys/pal/PLATFORM/thread.rs:LL:CC
+   = note: inside `std::thread::Builder::spawn_unchecked_::<'_, {closure@tests/fail-dep/concurrency/libc_pthread_mutex_deadlock.rs:LL:CC}, ()>` at RUSTLIB/std/src/thread/mod.rs:LL:CC
+   = note: inside `std::thread::Builder::spawn_unchecked::<{closure@tests/fail-dep/concurrency/libc_pthread_mutex_deadlock.rs:LL:CC}, ()>` at RUSTLIB/std/src/thread/mod.rs:LL:CC
+   = note: inside `std::thread::Builder::spawn::<{closure@tests/fail-dep/concurrency/libc_pthread_mutex_deadlock.rs:LL:CC}, ()>` at RUSTLIB/std/src/thread/mod.rs:LL:CC
+   = note: inside `std::thread::spawn::<{closure@tests/fail-dep/concurrency/libc_pthread_mutex_deadlock.rs:LL:CC}, ()>` at RUSTLIB/std/src/thread/mod.rs:LL:CC
```

Comment:
It's not pretty to show this for every backtrace. We spent a bunch of effort making backtraces nice, i.e. they usually stop at `main` and don't go all the way to the start lang item. We should do something similar here to avoid drowning the user in noise.

I could imagine two options:
- we mark these frames as `#[caller_location]` in std (we have done this for a bunch of other functions to make backtraces prettier)
- when pruning the stack for another thread, we also remove all non-local frames at the *top*

I am leaning towards the first option since the second option might prune too much when the thread is spawned deep inside some other library.

Reply:
I have posted https://github.com/rust-lang/rust/pull/134560 to add the `#[caller_location]` for this.